airflow/providers/oracle/operators/oracle.py

```diff
@@ -98,7 +102,9 @@ def execute(self, context: Context):
         try:
             return hook.callproc(self.procedure, autocommit=True, parameters=self.parameters)
         except oracledb.DatabaseError as e:
-            code, mesg = e.args[0].message[:-1].split(": ", 1)
-            if self.do_xcom_push:
-                ti.xcom_push(key="ORA", value=str(code.split("-")[1]))
+            code_match = re.search("^ORA-(\\d+):.+", str(e))
+            if code_match is not None:
+                if self.do_xcom_push:
+                    if ti is not None:
+                        ti.xcom_push(key="ORA", value=code_match.group(1))
```

Comment:
IMO, this is cognitively complex.

```suggestion
            if code_match and self.do_xcom_push and ti:
                ti.xcom_push(key="ORA", value=code_match.group(1))
```